nntrainer/src/databuffer.cpp

```diff
@@ -415,4 +416,50 @@ bool DataBuffer::getDataFromBuffer(BufferType type, std::vector<std::vector<std:
 
   return true;
 }
+
+int DataBuffer::setDataFile(std::string path, DataType type) {
+  int status = ML_ERROR_NONE;
+  std::ifstream data_file(path);
+  if (!data_file.good()) {
+    ml_loge("Error: Cannot open configuraiotn file %s", path.c_str());
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  switch (type) {
+    case DATA_TRAIN:
+      train_file = path;
+      break;
+    case DATA_VAL:
+      val_file = path;
+      break;
+    case DATA_TEST:
+      test_file = path;
+      break;
+    case DATA_LABEL: {
+      std::string data;
+      while (data_file >> data) {
+        labels.push_back(data);
+      }
+    }
+      if (labels.size() != class_num) {
+        ml_loge("Error: number of label is not equal to number of class : %lu vs. %d", labels.size(), class_num);
+        return ML_ERROR_INVALID_PARAMETER;
+      }
+      break;
+    default:
+      break;
```

Comment:
I think
1. It is better to return `ML_ERROR_INVALID_PARAMETER` in case of default since it is not handled in this code.
2. there should be `case DATA_UNKNOWN`  with `ml_loge` for better debugging

Reply:
fixed.